backend/src/Designer/TypedHttpClients/DelegatingHandlers/CachingDelegatingHandler.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Caching.Memory;
+
+namespace Altinn.Studio.Designer.TypedHttpClients.DelegatingHandlers
+{
+    public class CachingDelegatingHandler : DelegatingHandler
+    {
+        /// If needed add headers to the cache key
+        internal class CacheResponseDataEntry
+        {
+            public byte[] Data { get; set; }
+            public HttpStatusCode StatusCode { get; set; }
+        }
+
+        private readonly IMemoryCache _memoryCache;
+        private readonly int _cacheExpiryInSeconds;
+
+        public CachingDelegatingHandler(IMemoryCache memoryCache, int cacheExpiryInSeconds = 10 * 60)
```

Comment:
Any reason not to write 600? 
And should we use a constant to define the number so we know what the number represents?